### ISSUE TITLE
Proposal: move configuration to config directory

### ISSUE THREAD

**Issue by aykevl** [devgianlu/go-librespot#100]
Right now, go-librespot reads configuration and credentials from the current working directory (unless specified otherwise). I propose we use the following scheme instead:

* Configuration is read from ~/.config/go-librespot/config.yml. The file is the same as it currently is.
* Credentials are stored in a new state file, in ~/.config/go-librespot/state.json. The file looks like this:
    ```json
    {
      "credentials": {
        "username": "...",
        "data": "..."
      },
      "device_id": "..."
    }
    ```
* The config directory is configurable with a flag, defaulting to `UserConfigDir() + "/go-librespot"` so it also uses the appropriate directory on other OSes.

There are a few reasons why I think this is a good idea:

* I think the current directory is a bad place to store configuration. Maybe it's fine on a headless system, but at least the way I use go-librespot is on a regular user account (with config directory available).
* More state needs to be saved, such as [ZeroConfig credentials](https://github.com/devgianlu/go-librespot/issues/81). I also have an issue that Spotify keeps defaulting to "Let others Jam on this speaker" which I suspect is because the device ID is regenerated each time.

@devgianlu if you agree with this scheme I'd like to implement this (but asking here first before I put a lot of work into it).

**Comment by Willyarma** [comment]
Its more useful to me the way it is for my application, is there a way we can have both? perhaps with a command line option?

**Comment by aykevl** [comment]
@Willyarma yes, if there's a flag to set the config directory then you can set it to use the current working directory.

**Comment by Willyarma** [comment]
I already made a request in #80 for command line options. Its loosly related to this.
I was thinking about creating the config file dynamically on the fly then realised if I launch all instances in parallel then that wont work. A command line path to different config files would though.

**Comment by devgianlu** [comment]
@aykevl This sounds like a great idea, especially in tackling #81. I have recently refactored the configuration code to use `koanf` so it's a little bit nicer, but this is even better.

**Comment by aykevl** [comment]
Proposal is implemented in #103 so closing.
